model/internal/pdata/common.go

```diff
@@ -446,23 +446,23 @@ func newAttributeKeyValueBytes(k string, v []byte) otlpcommon.KeyValue {
 	return orig
 }
 
-// AttributeMap stores a map of attribute keys to values.
-type AttributeMap struct {
+// Map stores a map of attribute keys to values.
+type Map struct {
 	orig *[]otlpcommon.KeyValue
 }
 
-// NewAttributeMap creates a AttributeMap with 0 elements.
-func NewAttributeMap() AttributeMap {
+// NewMap creates a Map with 0 elements.
+func NewMap() Map {
 	orig := []otlpcommon.KeyValue(nil)
-	return AttributeMap{&orig}
+	return Map{&orig}
 }
 
-// NewAttributeMapFromMap creates a AttributeMap with values
+// NewMapFromRaw creates a Map with values
 // from the given map[string]AttributeValue.
-func NewAttributeMapFromMap(attrMap map[string]AttributeValue) AttributeMap {
+func NewMapFromRaw(attrMap map[string]AttributeValue) Map {
```

Comment:
This name is chosen for consistency with existing `Map.AsRaw` method

Reply:
The only difference between them is that `AsRaw` returns `map[string]interface{}` not `map[string]AttributeValue` but I think it's ok. The types shouldn't be changed given their use cases.

Reply:
Hmm... I don't like this inconsistency... let's think about this because it is a subtle difference that may confuse users.

Reply:
Actually I think we can change NewMapFromRaw to take `map[string]interface{}` as well. I checked it's being used in tests only, so performance overhead (if there is any) doesn't matterr. Let me know what do you think. It'll make client code even more concise. Let me know what do you think

Reply:
Would like to see that code..

Reply:
Made this change in the second commit to this PR. Please let me know what do you think